BorrowBookControl.java

```diff
@@ -0,0 +1,168 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class BorrowBookControl {
+
+    private BorrowBookUI bookUI;
+    private library library;
+    private member member;
+    private enum ControlState { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
+    private ControlState state;
+    private List<book> pending;
+    private List<loan> completed;
+    private book book;
+
+
+	public BorrowBookControl() {
+
+    this.library = library.INSTANCE();
+    this.state = ControlState.INITIALISED;
+
+	}
+
+
+	public void setUI(BorrowBookUI bookUI) {
+
+    if (!state.equals(ControlState.INITIALISED)) {
```

Comment:
DEFECT: Guideline custom: Reference to class members need to have the "this" qualifier